src/utils/url.js

```diff
@@ -0,0 +1,32 @@
+// @ts-check
+
+/**
+ *
+ * @param {URL} url
+ * @returns {URL|string}
+ */
+export function parseUrl (url) {
+  try {
```

Comment:
@guanzo had to add this try catch here because when testing in the browser, `process` is not defined. Seems like webpack's environment plugin doesn't make `process` accessible in libraries that you import from. This `parseUrl` function is only for testing though and is temporary so think this is fine for now.

Reply:
why's it being imported by the browser-client if its only used for testing

Reply:
sorry its not only for testing, parseUrl is used within the js-client code. Im was just saying the reason it exists because the mock tests don't accept `URL` objects in `fetch` so had to add this to make them run.